clang/lib/Format/ContinuationIndenter.cpp

```diff
@@ -1203,6 +1214,10 @@ unsigned ContinuationIndenter::addTokenOnNewLine(LineState &State,
     }
   }
 
+  if (!Style.BinPackBinaryOperations && Previous.is(TT_BinaryOperator) &&
+      (Previous.getPrecedence() > prec::Conditional)) {
+    CurrentState.BreakBeforeParameter = true;
```

Comment:
which test is for this condition

Reply:
This is for cases such as this:
```c++
result = operand1 +
         operand2 -
         operand3 +
         operand4 -
         operand5 +
         operand6;
```
without it, the formatting could do this:
```c++
result = operand1 +
         operand2 -
         operand3 + operand4 - operand5 + operand6;
```